# Group Project A Report

## Can We Predict the Presence of Heart Diseases from Observing Resting Blood Pressure and Serum Cholesterol? 

### Introduction:







### Exploratory Work on the Data

In [39]:
library(tidyverse)
library(tidymodels)
library(repr)
library(RColorBrewer)

In [25]:
heart_data <- read_csv("https://raw.githubusercontent.com/oskarjychen/dsci100_group_a/main/data/processed.cleveland.data",
                       col_names = FALSE)

selected_heart_data <- heart_data |>
    select(trestbps = X4, chol = X5, num = X14) 
selected_heart_data

Rows: 303 Columns: 14
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (2): X12, X13
dbl (12): X1, X2, X3, X4, X5, X6, X7, X8, X9, X10, X11, X14

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


trestbps chol num
1   145      233  0  
2   160      286  2  
3   120      229  1  
4   130      250  0  
5   130      204  0  
6   120      236  0  
7   140      268  3  
8   120      354  0  
9   130      254  2  
10  140      203  1  
11  140      192  0  
12  140      294  0  
13  130      256  2  
14  120      263  0  
15  172      199  0  
16  150      168  0  
17  110      229  1  
18  140      239  0  
19  130      275  0  
20  130      266  0  
21  110      211  0  
22  150      283  0  
23  120      284  1  
24  132      224  3  
25  130      206  4  
26  120      219  0  
27  120      340  0  
28  150      226  0  
29  150      247  0  
30  110      167  3  
⋮   ⋮        ⋮    ⋮  
274 112      149  0  
275 134      204  1  
276 170      227  0  
277 146      278  0  
278 138      220  0  
279 154      232  1  
280 130      197  0  
281 110      335  2  
282 130      253  0  
283 128      205  3  
284 122      192  0  
285 148      203  2  
286 114      318  4  
287 170      225  2  
288 125      220  0  
289 130      221  0  
290 120      240  0  
291 152      212  1  
292 132      342  0  
293 120      169  2  
294 140      187  2  
295 124      197  1  
296 120      157  0  
297 164      176  3  
298 140      241  1  
299 110      264  1  
300 144      193  2  
301 130      131  3  
302 130      236  1  
303 138      175  0

*figure 1*

Since the *num* variable is in essence categorical, we will creat a new binary column to signify whether heart disease in present in the patient. The new column will be called heart_disease, with the *num* value of 0 represented by "none", and the values of 1, 2, 3, 4 being represented by "present".

We will also be changing both *num* and *heart_disease* to factor columns:

In [35]:
clean_heart_data <- selected_heart_data |>
    mutate(heart_disease = ifelse(num > 0, "present", "none")) |>
    mutate(num = as_factor(num), heart_disease = as_factor(heart_disease))
clean_heart_data

trestbps chol num heart_disease
1   145      233  0   none         
2   160      286  2   present      
3   120      229  1   present      
4   130      250  0   none         
5   130      204  0   none         
6   120      236  0   none         
7   140      268  3   present      
8   120      354  0   none         
9   130      254  2   present      
10  140      203  1   present      
11  140      192  0   none         
12  140      294  0   none         
13  130      256  2   present      
14  120      263  0   none         
15  172      199  0   none         
16  150      168  0   none         
17  110      229  1   present      
18  140      239  0   none         
19  130      275  0   none         
20  130      266  0   none         
21  110      211  0   none         
22  150      283  0   none         
23  120      284  1   present      
24  132      224  3   present      
25  130      206  4   present      
26  120      219  0   none         
27  120      340  0   none         
28  150      226  0   none         
29  150      247  0   none         
30  110      167  3   present      
⋮   ⋮        ⋮    ⋮   ⋮            
274 112      149  0   none         
275 134      204  1   present      
276 170      227  0   none         
277 146      278  0   none         
278 138      220  0   none         
279 154      232  1   present      
280 130      197  0   none         
281 110      335  2   present      
282 130      253  0   none         
283 128      205  3   present      
284 122      192  0   none         
285 148      203  2   present      
286 114      318  4   present      
287 170      225  2   present      
288 125      220  0   none         
289 130      221  0   none         
290 120      240  0   none         
291 152      212  1   present      
292 132      342  0   none         
293 120      169  2   present      
294 140      187  2   present      
295 124      197  1   present      
296 120      157  0   none         
297 164      176  3   present      
298 140      241  1   present      
299 110      264  1   present      
300 144      193  2   present      
301 130      131  3   present      
302 130      236  1   present      
303 138      175  0   none

*figure 2*

Now we can look at how the presence of heart diseases in this data frame is distributed:

In [37]:
num_obs <- nrow(clean_heart_data)
clean_heart_data |>
    group_by(heart_disease) |>
    summarize(count = n(), percentage = n()/num_obs * 100)

heart_disease count percentage
1 none          164   54.12541  
2 present       139   45.87459

*figure 3*

We can see that patients with no heart disease account for 54.13%, while those with heart disease make up for around 45.87%